scripts/change_did_key.js

```diff
@@ -0,0 +1,150 @@
+/// This script assumes that all keys are sr25519.
+///
+/// Warning: This script only supports sr25519 keys. Do not use it to set an non-sr25519 public key.
+///          If you do, you will lose control of your DID.
+
+import { Keyring } from '@polkadot/api';
+import { cryptoWaitReady } from '@polkadot/util-crypto';
+import { u8aToHex, assert } from '@polkadot/util';
+import { createSignedKeyUpdate } from '../src/utils/did';
+import { getPublicKeyFromKeyringPair } from '../src/utils/misc';
+import { DockAPI, PublicKeySr25519 } from '../src/api';
+import b58 from 'bs58';
+
+require('dotenv').config();
+
+/// Use (with node running locally):
+///
+/// ```bash
+/// cd sdk
+/// yarn
+/// npx babel-node ./scripts/change_did_key.js args..
+/// ```
+
+const USAGE = `\
+npx babel-node ./scripts/change_did_key.js <did> <new_pk>
+  where
+    <did> is a 32 byte, 0x prefixed hex string.
```

Comment:
You don't need to require 32 bytes. The places where this is used, i.e. in `createSignedKeyUpdate` and `getSr25519PkHex` both accept hex or qualified like `did:dock:..`. You can use `getHexIdentifierFromDID` to validate the script arg for that. People creating a DID will usually have a qualified did.

Reply:
Used `getHexIdentifierFromDID` to allow fully qualified dids. 754c03c

Instead of propagating the possibly fully qualified DID deeper into the program I parse it using `getHexIdentifierFromDID` and pass on the parsed value. Behavior is the same.